src/app/core/cart/cart.service.ts

```diff
@@ -61,12 +66,16 @@ export class CartService {
 
     private evaluateCartWeight(items: any): number {
         return items
-            .filter((item: any) => item?.dimensions?.weight != null && item?.quantity != null)
-            .reduce((sum: number, item: any) => sum + (item.dimensions.weight * item.quantity), 0);
+            .map((item: any) => item.quantity * this.getSnipcartItemWeight(item))
+            .reduce((sum: number, weight: number) => sum + weight, 0);
```

Comment:
both map and reduce loop through the whole array. Chaining both results in O(2n) complexity.

We can do all the logic within a single reduce block to limit the complexity to O(n)